### DISCUSSION TITLE
Detect theme light or dark mode?

### DISCUSSION THREAD

**Discussion by mkosir** [facebook/docusaurus#9196] [Q&A] [answered]
Pretty sure it's possible detecting in React components theme light/dark mode, but can't find info on it.
As I can see `useThemeContext` hook is not in v2 anymore 🤔 ? Should we use `useDocusaurusContext` hook?

**Comment by slorber** [comment, marked as answer]
https://docusaurus.io/docs/api/themes/configuration#use-color-mode (probably not documented in the right place)

I don't recommend using `useColorMode()` because the value is known only after SSG + React hydration: the initial client-side value on the very first render will be wrong (always the default theme value). 

It is usually better to use the CSS selectors instead, to avoid a "light/dark mode flash":
https://docusaurus.io/docs/styling-layout#dark-mode

**Comment by mkosir** [comment]
@slorber thank you 👍
